paymentsheet/src/test/java/com/stripe/android/paymentsheet/DefaultIntentConfirmationInterceptorTest.kt

```diff
@@ -519,6 +521,67 @@ class DefaultIntentConfirmationInterceptorTest {
         )
     }
 
+    @Test
+    fun `If requires next action with an attached payment method different then the created one, throw error`() =
+        runTest {
+            val paymentMethod = PaymentMethodFixtures.CARD_PAYMENT_METHOD
+
+            val interceptor = DefaultIntentConfirmationInterceptor(
+                stripeRepository = object : AbsFakeStripeRepository() {
```

Comment:
I think this would be more clear if we moved it into a function

`stripeRepositoryReturning(onCreate = "pm_1", onRetrieve = "pm_2)`